primer/test/Tests/Action/Available.hs

```diff
@@ -333,14 +334,36 @@ tasty_available_actions_accepted = withTests 500 $
                       , case astTypeDefParameters def' of
                           [] -> pure forTypeDef
                           ps -> do
-                            (p, _) <- Gen.element ps
-                            pure
-                              ( "forTypeDefParamNode"
+                            (p, k) <- Gen.element ps
+                            let typeDefParamNodeSel = typeDefSel . Just . TypeDefParamNodeSelection . TypeDefParamSelection p
+                            Gen.frequency
+                              [
+                                ( 1
+                                , pure
+                                    ( "forTypeDefParamNode"
+                                    ,
+                                      ( typeDefParamNodeSel Nothing
+                                      , Available.forTypeDefParamNode p l defMut allTypes' allDefs' defName def'
+                                      )
+                                    )
+                                )
                               ,
-                                ( typeDefSel $ Just $ TypeDefParamNodeSelection p
-                                , Available.forTypeDefParamNode p l defMut allTypes' allDefs' defName def'
+                                ( 3
+                                , do
+                                    let allKindIDs = \case
+                                          KHole m -> [getID m]
+                                          KType m -> [getID m]
+                                          KFun m k1 k2 -> [getID m] <> allKindIDs k1 <> allKindIDs k2
```

Comment:
This is another thing which can be simplified once we have zippers etc. for kinds.